dns_entry_checker/test/test_dns_entry_checker.py

```diff
@@ -0,0 +1,193 @@
+
+        with patch("dns_entry_checker.ssh_command"):
+            dns_entry_checker.ssh_command.return_value = returned_dns
+
+            check_ip_dns_mismatch(
+                ips_dns_pair,
+                client,
+                backward_mismatch_filepath,
+                forward_mismatch_filepath,
+                backward_missing_filepath,
+            )
+
+        if not expected_out:
+            assert call("test/backward/missing/filepath", "a") in mock_file.mock_calls
+        else:
+            assert (
+                call("test/backward/missing/filepath", "a") not in mock_file.mock_calls
+            )
+
+    @parameterized.expand(
+        [
+            ("Return doesn't contain IP", ["No IPS here"], 0),
+            ("Returned IP doesn't match", ["test-host-172-16-1-2.nubes.stfc.ac.uk"], 1),
+            ("Returned IP matches", ["test-host-172-16-1-1.nubes.stfc.ac.uk"], 2),
+        ]
+    )
+    @mock.patch("builtins.open")
+    def test_check_ip_dns_mismatch_backwards_not_found(
+        self,
+        name,
+        returned_ips,
+        expected_out,
+        mock_file,
+    ):
+        ips = ["172-16-1-1", "172.16.1.1"]
+        client = MagicMock()
+        backward_mismatch_filepath = "test/backward/mismatch/filepath"
+        forward_mismatch_filepath = "test/forward/mismatch/filepath"
+        backward_missing_filepath = "test/backward/missing/filepath"
+
+        with patch("dns_entry_checker.ssh_command"):
+            dns_entry_checker.ssh_command.return_value = returned_ips
+
+            check_ip_dns_mismatch(
+                ips,
+                client,
+                backward_mismatch_filepath,
+                forward_mismatch_filepath,
+                backward_missing_filepath,
+            )
+
+        if expected_out == 0:
+            assert call("test/backward/missing/filepath", "a") in mock_file.mock_calls
+        elif expected_out == 1:
+            assert call("test/backward/mismatch/filepath", "a") in mock_file.mock_calls
+        else:
+            assert (
+                call("test/backward/mismatch/filepath", "a") not in mock_file.mock_calls
+            )
+            assert (
+                call("test/backward/missing/filepath", "a") not in mock_file.mock_calls
+            )
+
+    @parameterized.expand(
+        [
+            ("Gaps", [5], True),
+            ("No gaps", sorted(set(range(2, 255))), False),
+        ]
+    )
+    @mock.patch("builtins.open")
+    def test_check_missing_ips(self, name, ips, expected_out, mock_file):
+        key = ("105", [str(x).zfill(0) for x in ips])
+        gap_missing_filepath = "test/gap/missing/filepath"
+
+        check_missing_ips(key, gap_missing_filepath)
+
+        print(expected_out, mock_file.mock_calls)
+
+        if expected_out:
+            assert call("test/gap/missing/filepath", "a") in mock_file.mock_calls
+        else:
+            assert call("test/gap/missing/filepath", "a") not in mock_file.mock_calls
+
+
```

Comment:
you're missing tests for check_output_files and populate_ip_dict

i'd recommend using `coverage` python tool to see if you covered all tests
https://www.pythontutorial.net/python-unit-testing/python-unittest-coverage/